stout/flags.cc

```diff
@@ -101,6 +188,34 @@ void Parser::Parse(int* argc, const char*** argv) {
 
     // Skip anything that doesn't look like a flag.
     if (arg.find("--") != 0) {
+      if (subcommands_.count(arg) > 0) {
+        std::cerr << "Encountered duplicate subcommand '"
+                  << arg << "'." << std::endl;
+        exit(1);
+      }
+      if (previous_subcommand_message_
+                  .value()
+                  ->GetDescriptor()
+                  ->FindFieldByName(arg)
+              != nullptr
+          && parsing_subcommand_flags_) {
+        std::cerr << "You have already set oneof 'subcommand'"
+                  << " field for the message '"
+                  << previous_subcommand_message_.value()->GetTypeName()
+                  << "'" << std::endl;
+        exit(1);
+      }
+      // It could be probably subcommand.
+      if (google::protobuf::Message* subcommand_message =
+              TryParseSubcommand(arg);
+          subcommand_message != nullptr) {
+        parsing_subcommand_flags_ = true;
+        previous_subcommand_message_ = cur_subcommand_message_;
+        cur_subcommand_message_ = subcommand_message;
+        subcommands_.insert(arg);
+        AddAllOrExit(*cur_subcommand_message_);
```

Comment:
Instead of doing this here how about we do this when we first create the parser and we create _nested_ parsers for each subcommand so that we can just call into that nested parser here! That will allow there to be nested flags with the same name.